### ISSUE TITLE
Issue with parsing pipe symbol inside of the cell

### ISSUE THREAD

**Issue by ext-rhorodechnyi** [mapado/prettier-plugin-gherkin#19]
We have faced an issue when the table cell contains text with the pipe symbol `|`.
According to gherkin documentation, we are trying to escape it `\|` but prettier removes `\` end we are getting an error
`Row must have the same cells count as table's header `
Code example:
```
| element                  | should           | value                                                       |
| all select fields list   | have length      | 5                                                           |
| select audit             | contain text     | 79636 \| B - 10/10/2017 \| some social audit description #1 |
```
Prettier version `3.2.5`

**Comment by jdeniau** [comment]
Hi,

This may be a known issue, I think that you can read the documentation about [escapeBackslashes](https://github.com/mapado/prettier-plugin-gherkin?tab=readme-ov-file#escapebackslashes) : there is a difference between the behat parser and the gherkin documentation and the cucumber implementation.

To be honest, I do not remember exactly what you should do in your case, but you may want to set it to `true` or try to escape your pipe twice : `\\|`.

**Comment by zciluk** [comment]
@ext-rhorodechnyi 
> We have faced an issue when the table cell contains text with the pipe symbol `|`. According to gherkin documentation, we are trying to escape it `\|` but prettier removes `\` end we are getting an error `Row must have the same cells count as table's header ` Code example:
> 
> ```
> | element                  | should           | value                                                       |
> | all select fields list   | have length      | 5                                                           |
> | select audit             | contain text     | 79636 \| B - 10/10/2017 \| some social audit description #1 |
> ```
> 
> Prettier version `3.2.5`

were you able to find a solution to this? 
I have a similar issue where escaping **multiple** pipe symbols in one field: 
```
| firstColumn  | secondColumn | value                    |
| foo          | bar          | a\| b\| c \|             |
```
is being formatted to: 
```
| firstColumn  | secondColumn  | value               |
| foo          | bar           | a\| b| c|           |
```
and with _escapeBackslashes_ set to true it just changes to `a\\| b| c| `. 
So after formatting, running it again leaves error `inconsistent cell count within the table`, because of unescaped pipes. 

Escaping with double backslash like this `a\\| b\\| c \\| ` leave me also with inconsistent cell count error. 
How can multiple pipe symbols be escaped then?

**Comment by jdeniau** [comment]
Sorry, I did miss the point where several pipes are escaped ! 
I'm looking at it right now.

**Comment by jdeniau** [comment]
The muli-escaped pipes bug has been fixed in 3.1.0.
